Modules/_io/fileio.c

```diff
@@ -418,7 +421,12 @@ _io_FileIO___init___impl(fileio *self, PyObject *nameobj, const char *mode,
 
         fd_is_own = 1;
         if (self->fd < 0) {
+#ifdef MS_WINDOWS
+            PyErr_SetExcFromWindowsErrWithFilenameObject(
+                    PyExc_OSError, _doserrno, nameobj);
```

Comment:
What does the error look like in the path too long case with this change? Would it be more helpful if we checked the error code and used our own message?